digiwf-integrations/digiwf-dms-integration/digiwf-dms-integration-core/src/main/java/de/muenchen/oss/digiwf/dms/integration/adapter/out/s3/S3Adapter.java

```diff
@@ -66,11 +67,16 @@ private Content getFile (String filepath) {
             final String type = tika.detect(bytes);
             final String filename = FilenameUtils.getBaseName(filepath);
 
-            if(!supportedExtensions.contains(type.toLowerCase())) {
+            if(!supportedExtensions.containsValue(type.toLowerCase())) {
                 throw new BpmnError("FILE_TYPE_NOT_SUPPORTED", "The type of this file is not supported: " + filepath);
             }
 
-            return new Content(type, filename, bytes);
+            String extension = "";
+            for(Map.Entry<String, String> entry : supportedExtensions.entrySet()){
+                if(entry.getValue().equals(type)){extension = entry.getKey();}
+            }
+
```

Comment:
Meiner Meinung nach kann man die if Verzweigung und die for each schleife mehr funktional lösen: 

```
            final String extension = supportedExtensions.entrySet()
                    .stream()
                    .filter(set -> set.getValue().equals(type))
                    .findFirst()
                    .map(Map.Entry::getKey)
                    .orElseThrow(() -> new BpmnError("FILE_TYPE_NOT_SUPPORTED", "The type of this file is not supported: " + filepath));
```

Die Tests laufen zu mindestens noch durch. :-D 

Ist aber kein Blocker. :-)